frontend/src/pages/[username]/[projectName]/[multiProjectName]/index.js

```diff
@@ -0,0 +1,87 @@
+import React from 'react'
+
+import {
+  canCommit,
+  getDefaultBranchFiles,
+  getRepo,
+  repoExists,
+} from '@utils/giteaApi'
+import { findReadme, renderReadme } from '@utils/index'
+import {
+  getBoardBomInfo,
+  getBoardGerberInfo,
+  getKitspaceYAMLJson,
+  hasInteractiveBom,
+  processedAssets,
+} from '@utils/projectPage'
+import SharedProjectPage from '@components/SharedProjectPage'
+
+export const getServerSideProps = async ({ params, query, req }) => {
+  const processorUrl = process.env.KITSPACE_PROCESSOR_URL
+  // `repoFullname` is resolved by matching its name against the `page` dir.
+  // Then it's used to access the repo by the Gitea API.
+  const { multiProjectName, username, projectName } = params
+  const repoFullname = `${username}/${projectName}`
+
+  const kitspaceYAMLPath = `${processorUrl}/files/${repoFullname}/HEAD`
+  const assetsPath = `${processorUrl}/files/${repoFullname}/HEAD/${multiProjectName}`
+
+  // The repo owner and collaborators can upload files.
+  const hasUploadPermission = await canCommit(
+    repoFullname,
+    req?.session?.user?.username,
+  )
+
+  if (await repoExists(repoFullname)) {
+    const repo = await getRepo(repoFullname)
+    const repoFiles = await getDefaultBranchFiles(repoFullname)
+
+    // TODO: ALL assets aren't available for the repos which are being processed,
+    // or the repos that don't have assets from first place.
+    // This should be handled properly currently, it breaks the page.
```

Comment:
So it's broken when making a new project right now?

Reply:
nit: "from the first place" -> "in the first place"

Reply:
Resolved in b83ddbb.